### PR TITLE
Remove list's setFilters default debounce

### PR DESCRIPTION
**Pull request by fzaninotto**
## Problem

The `setFilters` callback returned by `useListController` is debounced by default. This creates numerous WTF issues (e.g. #4189). This default makes sense for the Filter form/button combo, but not for the other filters.

Besides, the `setFilters` returned by other controllers (`useList`, etc) isn’t debounced by default.

## Solution

Make the `setFilters` default to `debounce=false`. Force the debounce in `FilterForm`.

Also, revert #9639 as these explanations are no longer necessary

### CODE REVIEW

File: packages/ra-ui-materialui/src/input/AutocompleteInput.tsx [hunk 1]
```diff
@@ -345,7 +345,7 @@ If you provided a React element for the optionText prop, you must also provide t
                 return;
             }
 
-            setFilters(filterToQuery(filter), undefined, true);
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/AutocompleteInput.tsx:348]: This didn't make sense as the call is debounced two lines earlier. The previous code caused a double debounce.

File: docs/FilteringTutorial.md [hunk 2]
```diff
@@ -543,14 +543,14 @@ const PostFilterForm = () => {
 
     const onSubmit = (values) => {
         if (Object.keys(values).length > 0) {
-            setFilters(values, undefined, false);
+            setFilters(values, undefined);
```
Comment (slax57) [inline docs/FilteringTutorial.md:546]: **polish:** ‏You can probably simply this even further

```suggestion
            setFilters(values);
```

File: docs/useListContext.md [hunk 3]
```diff
@@ -142,8 +142,7 @@ const CustomFilter = () => {
 
     const handleSubmit = (event) => {
         event.preventDefault();
-        // The 3rd parameter disables the debounce ⤵
-        setFilters(filterFormValues, undefined, false);
+        setFilters(filterFormValues, undefined);
```
Comment (slax57) [inline docs/useListContext.md:145]: **polish:** ‏same

```suggestion
        setFilters(filterFormValues);
```

File: docs/useListController.md [hunk 4]
```diff
@@ -282,8 +282,7 @@ const OfficeList = () => {
 
     const handleSubmit = (event) => {
         event.preventDefault();
-        // The 3rd parameter disables the debounce ⤵
-        setFilters(filterFormValues, undefined, false);
+        setFilters(filterFormValues, undefined);
```
Comment (slax57) [inline docs/useListController.md:285]:
```suggestion
        setFilters(filterFormValues);
```

File: packages/ra-ui-materialui/src/input/AutocompleteInput.tsx [hunk 5]
```diff
@@ -345,7 +345,7 @@ If you provided a React element for the optionText prop, you must also provide t
...
+            setFilters(filterToQuery(filter), undefined);
```
Comment (slax57) [inline packages/ra-ui-materialui/src/input/AutocompleteInput.tsx:348]:
```suggestion
            setFilters(filterToQuery(filter));
```

File: packages/ra-ui-materialui/src/list/filter/FilterListItem.tsx [hunk 6]
```diff
@@ -164,7 +164,7 @@ export const FilterListItem = memo((props: FilterListItemProps) => {
     const isSelected = getIsSelected(value, filterValues);
 
     const handleClick = () =>
-        setFilters(toggleFilter(value, filterValues), null, false);
+        setFilters(toggleFilter(value, filterValues), null);
```
Comment (slax57) [inline packages/ra-ui-materialui/src/list/filter/FilterListItem.tsx:167]:
```suggestion
        setFilters(toggleFilter(value, filterValues));
```

File: packages/ra-ui-materialui/src/list/filter/FilterLiveSearch.tsx [hunk 7]
```diff
@@ -36,10 +36,14 @@ export const FilterLiveSearch = memo((props: FilterLiveSearchProps) => {
...
-            setFilters({ ...filterValues, [source]: event.target.value }, null);
+            setFilters(
+                { ...filterValues, [source]: event.target.value },
+                null,
+                true
+            );
         } else {
             const { [source]: _, ...filters } = filterValues;
-            setFilters(filters, null, false);
+            setFilters(filters, null);
```
Comment (slax57) [inline packages/ra-ui-materialui/src/list/filter/FilterLiveSearch.tsx:46]:
```suggestion
            setFilters(filters);
```

**Review by slax57 — COMMENTED** [review]
Some nitpicking. Otherwise, LGTM!

**Review by adguernier — APPROVED** [review]

**Review by slax57 — APPROVED** [review]

**Pull request merged**